### PR TITLE
Build linux version of the TrailDB library from sources

### PR DESCRIPTION
**Pull request by mrjoe7**
This change will build TrailDB from source. Right now it's working for linux only.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "traildb"]
+	path = traildb
+	url = https://github.com/traildb/traildb.git
```
Comment (19h) [inline .gitmodules]: sure you want this to be a submodule? seems rather unintuitive.

File: .travis.yml [hunk 2]
```diff
@@ -1 +1,27 @@
...
+      - clang
+      - libjudy-dev
+      - libarchive-dev
+      - pkg-config
+      - build-essential
+      - m4
+
+before_install:
+  - git clone https://github.com/traildb/traildb /tmp/traildb
+  - pushd /tmp/traildb && curl -o ./waf https://waf.io/waf-2.0.18 && chmod +x ./waf && ./waf configure && ./waf build && sudo ./waf install && sudo ldconfig && popd
```
Comment (19h) [inline .travis.yml]: I'd move the waf version into an isolated environment variable, but it's probably a thing of taste.

File: build.rs [hunk 3]
```diff
@@ -13,3 +11,70 @@ fn main() {
...
+    compiler.flag_if_supported("-Wshadow");
+    compiler.flag_if_supported("-Wstrict-prototypes");
+
+    compiler.cpp(false);
+    compiler.compile("traildb");
+}
+
+fn dep_libs() {
+    // Judy does not support querying by the pkg-config
+    println!("cargo:rustc-link-lib=dylib=Judy");
```
Comment (19h) [inline build.rs:18]: is this supposed to be a capital J?

Comment (mrjoe7) [reply build.rs:18]: It's kinda weird, but yes
```
tomas@localhost:~> whereis libJudy
libJudy: /usr/lib64/libJudy.so
```

File: .gitmodules (hunk 1 above)
Comment (mrjoe7) [reply .gitmodules]: At first I thought the same about using git submodules, but after looking at other crates I have found that mainly two approaches are used to deal with this:
- use git submodules
- commit C sources (or tar with sources) to git repository

When using the 1st approach, users of `traildb-rust` won't even notice that git submodules are used. The cargo will checkout git submodule automatically during the build process. Only downside of this approach is that the developers of `traildb-rust` will have to run `git submodule update --init --recursive` if they need to build traildb library from sources. If they have traildb already installed as a system library, then they don't need to do it at all.

File: .travis.yml (hunk 2 above)
Comment (mrjoe7) [reply .travis.yml]: good idea, pushed additional commit

File: .gitmodules (hunk 1 above)
Comment (mrjoe7) [reply .gitmodules]: I have moved traildb build into a separate package + traildb sources are now part of the repo.

**Comment by 19h** [comment]
Great, thanks for your changes!

**Review by 19h — APPROVED** [review]

**Comment by 19h** [comment]
hey @mrjoe7

I took some time to rewrite significant parts of the build process and worked off your state. I'm now generating the bindings on the fly in the traildb-sys crate and using it to expose the ffi to the main crate.

All changes are live in [traildb v0.6.0](https://crates.io/crates/traildb) & [traildb-sys  0.6.3](https://crates.io/crates/traildb-sys).

Please give it a shot & thanks for your contributions!

**Comment by mrjoe7** [comment]
hi @19h 

I have tried the new code and it looks good, nice work!

**Pull request merged**